src/main/java/net/javadiscord/javabot/api/routes/JDAEntity.java

```diff
@@ -0,0 +1,14 @@
+package net.javadiscord.javabot.api.routes;
+
+import net.dv8tion.jda.api.JDA;
+import net.javadiscord.javabot.Bot;
+
+/**
+ * Simple interface which adds the {@link JDAEntity#getJDA()} method which
+ * returns the bots' {@link JDA} client.
+ */
+public interface JDAEntity {
+	default JDA getJDA() {
```

Comment:
Make JDA a Spring Bean by creating a method like this in any Spring managed component (preferable an [`@Configuration` class](https://docs.spring.io/spring-javaconfig/docs/1.0.0.M4/reference/html/ch02s02.html)):
```java
@Bean
public void getJDA(){
	return Bot.getDih4jda().getJDA();
}
```
Spring will then discover that bean and allow to autowire the JDA object.

Spring controllers could then look like this:
```java
@RestController
@RequiredArgsConstructor
public class SomeController{
	private final JDA jda;
}
```

Aside from that, creating interfaces for a accessing methods and constants is an antipattern.